src/Compilers/Core/MSBuildTask/DiscoverEditorConfigFiles.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    /// <summary>
+    /// Locates the applicable set of .editorconfig files based on given inputs. Walks through
+    /// parent directories locating the ones that exist.
+    /// </summary>
+    public sealed class DiscoverEditorConfigFiles : Task
+    {
+        /// <summary>
+        /// The set of source and other files that we should find the applicable .editorconfig files to.
+        /// </summary>
+        [Required]
+        public ITaskItem[] InputFiles { get; set; }
+
+        [Output]
+        public ITaskItem[] EditorConfigFiles { get; private set; }
+
+        static DiscoverEditorConfigFiles()
+        {
+            AssemblyResolution.Install();
+        }
+
+        public DiscoverEditorConfigFiles()
+        {
+            TaskResources = ErrorString.ResourceManager;
+        }
+
+        public override bool Execute()
+        {
+            try
+            {
+                ExecuteCore();
+                return !Log.HasLoggedErrors;
+            }
+            catch (Exception ex)
+            {
+                // We'll explicitly log friendly errors and then rethrow in the case of IO errors while reading the file system.
```

Comment:
Unless LogErrorFromException throws, there doesn't appear to be any rethrow here?